server/routes/receiveData/index.js

```diff
@@ -1,10 +1,487 @@
+
+var AttitudeImuAccelerometer = require('./../../database/models/dataAttitude/attitudeImuAccelerometer.js')
+var AttitudeImuGyroscope = require('./../../database/models/dataAttitude/attitudeImuGyroscope.js')
+var AttitudeImuMagnetometer = require('./../../database/models/dataAttitude/attitudeImuMagnetometer.js')
+var AttitudeIrSensor = require('./../../database/models/dataAttitude/attitudeIrSensor.js')
+var AttitudeMagnetometer = require('./../../database/models/dataAttitude/attitudeMagnetometer.js')
+var AttitudePhotoDiode = require('./../../database/models/dataAttitude/attitudePhotoDiode.js')
+
+var FlashBurstLifepoCurrent = require('./../../database/models/dataFlashBurst/flashBurstLifepoCurrent.js')
+var FlashBurstLifepoVoltage = require('./../../database/models/dataFlashBurst/flashBurstLifepoVoltage.js')
+var FlashBurstTemperature = require('./../../database/models/dataFlashBurst/flashBurstTemperature.js')
+
+var RadioTransmissionCurrent = require('./../../database/models/dataRadioTransmission/radioTransmissionCurrent.js')
+var RadioTransmissionLionVoltage = require('./../../database/models/dataRadioTransmission/radioTransmissionLionVoltage.js')
+var RadioTransmissionTemperature = require('./../../database/models/dataRadioTransmission/radioTransmissionTemperature.js')
+
 // Main route for receiving data
 router.post('/', function(req, res, next){
+	saveAll(req.body, res)
+});
 
+function saveAll(rawJSON, res) {
+	var tid;
+	saveRaw(rawJSON)
+	.then(function(obj){
+		tid = obj.tid
+		return savePreamble(rawJSON.preamble, tid)
+	})
+	.then(function() {
+		return saveErrorInfo(rawJSON.error_info, tid)
+	}) 
+	.then(function() {
+		return saveCurrentData(rawJSON.current_data_info, tid)
+	})
+	.then(function() {
+		return saveAttitudeData(rawJSON.data_attitude, tid)
+	})
+	.then(function() {
+		return saveDataFlashBurst(rawJSON.data_flash_burst, tid)
+	})
+	.then(function() {
+		return saveDataRadioTransmission(rawJSON.data_radio_transmission, tid)
+	})
+	.then(function(){
+		return res.sendStatus(200)
+	})
+	.catch(function(error){
+		console.error(error)
+		return res.sendStatus(500)
+	});
+}
 
-});
+function saveRaw(rawJSON) {
+	return (Raw.build({
+		raw: JSON.stringify(rawJSON)
+	}).save())
+}
+
+function savePreamble(preamble, tid) {
+	return (Preamble.build({
+		callsign: preamble.callsign,
+		timestamp: preamble.timestamp,
+		messageAndOpStates: preamble.message_and_op_states,
+		bytesInError: preamble.bytes_in_error,
+		bytesInMessage: preamble.bytes_in_message,
+		tid: tid
+	}).save())
+}
+
+function saveErrorInfo(errorInfo, tid) {
+	var promises = []
+	var errorArray = errorInfo.error
+	for(var i = 0; i < errorArray.length; i++){
+		promises.push(ErrorInfo.build({
+			index: i,
+			errorCode: errorArray[i],
+			tid: tid
+		}).save())
+	}
+	return Promise.all(promises)
+}
+
+function saveCurrentData(currentData, tid) {
+	var promises = []
+	
+	// Battery Data
```

Comment:
so here the `promises` array should not be passed into those functions but rather have a promise chain of all of those functions.
so it would look like

```
saveCurrBatteryChargingAnalogVoltage(...args)
.then(function() {
return saveCurrBatteryChargingDigitalSignal(...args)
})

```